MultiSource/Benchmarks/Olden/em3d/em3d.h

```diff
@@ -30,28 +34,33 @@ extern int local_p; /* percentage of local edges */
 
 #define assert(a) if (!a) {printf("Assertion failure\n"); exit(-1);}
 
+// CHECKED C: Here we cannot reason about the length of the array_ptrs for
+// coeffs and from_values because array_ptrs in structs do not yet work.
+// CHECKED C: Here we cannot reason about the length of the array_ptr for 
+// to_nodes because that information is not kept in this struct (it's kept
+// in a global).
 typedef struct node_t {
-  double *value;
-  struct node_t *next;
-  struct node_t **to_nodes; /* array of nodes pointed to */
-  double **from_values; /* array of ptrs to vals where data comes from */
-  double *coeffs; /* array of coeffs on edges */
+  ptr<double> value;
+  ptr<struct node_t> next;
+  ptr<struct node_t>* to_nodes : itype(array_ptr<ptr<struct node_t>>); /* array of nodes pointed to */
+  ptr<double> *from_values : itype(array_ptr<ptr<double>>); /* array of ptrs to vals where data comes from */
+  double *coeffs : itype(array_ptr<double>); /* array of coeffs on edges */
```

Comment:
I'm also not happy with these declarations, but Microsoft/checkedc-clang#244 prevents us from having the right array_ptr declaration.